### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry): define immersions

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/Morphisms/Immersion.lean [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    exact f.isLocallyClosed_range.image g.embedding.toInducing g.isLocallyClosed_range
+
+instance comp {X Y Z : Scheme} (f : X ⟶ Y) (g : Y ⟶ Z) [IsImmersion f]
+    [IsImmersion g] : IsImmersion (f ≫ g) :=
+  MorphismProperty.IsStableUnderComposition.comp_mem f g inferInstance inferInstance
+
+variable {f} in
+/--
+A morphism is a (locally-closed) immersion if and only if it can be factored into
+a closed immersion followed by a (dominant) open immersion.
```
Comment (joelriou) [inline Mathlib/AlgebraicGeometry/Morphisms/Immersion.lean:133]: In this lemma, we do not use this condition:
```suggestion
a closed immersion followed by an open immersion.
```

**Comment by joelriou** [comment]
Otherwise, it looks very good to me!

File: Mathlib/AlgebraicGeometry/Morphisms/Immersion.lean (hunk 1 above)
Comment (acmepjz) [reply Mathlib/AlgebraicGeometry/Morphisms/Immersion.lean:133]: Does it mean `locally-closed => dominant`?

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**